bootcamp-api.Tests/Services/BookmarkServiceTests.cs

```diff
@@ -5,9 +5,23 @@ public class BookmarkServiceTests: IDisposable
 
     private readonly DbConnection _connection;
     private readonly DbContextOptions<PawssierContext> _contextOptions;
+    private readonly IMapper _mapper;
+    private readonly int user_id = 1;
+    private readonly ApiVersion apiVersion = new ApiVersion(1, 0);
+    private readonly int pfVersion = 2;
 
     public BookmarkServiceTests()
```

Comment:
All info within this constructor should really be in a Setup method via the unit testing platform. A method that runs before every test. Otherwise you're running tests against data that changes every test. This means your tests become reliant on all running together.

The mapper is probably ok to be in the constructor though.

Generally we want to avoid any data related tests, otherwise they become more of an integration test (something that tests multiple systems or services). The unit test goal is to test a single thing, so we typically mock or stub out any dependencies we aren't concerned with in the context of the test. We care much more about testing conditions or different code paths.

Reply:
I guess this is just how xunit works. I didn't realize it used the constructor as the setup method.